_plugins/joiner.rb

```diff
@@ -139,17 +139,36 @@ def self.flatten_index(index)
       index.values
     end
 
+    def collect_data_from_team_collection
+      data = {}
+      collection = @site.config['team_collection']
+      @site.collections[collection].docs.each do |doc|
+        # Trim off the team collection and the file name from the path.
+        dirs = doc.relative_path.split(File::SEPARATOR)[1..-2]
+        data_class = dirs.pop
+        current = data
+        dirs.each {|dir| current = (current[dir] ||= Hash.new)}
+        (current[data_class] ||= Array.new) << doc.data
+        doc.data['title'] = doc.data[doc.data['title_field']]
+      end
+
+      if @public_mode
+        HashJoiner.remove_data data, 'private'
+      else
+        HashJoiner.promote_data data, 'private'
+      end
+
+      data.each {|k,v| @site.data[k] = v}
```

Comment:
Minor: you might be able to do `@site.data.merge!(data)`.

http://ruby-doc.org/core-2.2.0/Hash.html#method-i-merge-21